File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.cpp
class Solution {
public:
    long long getDescentPeriods(vector<int>& prices) {
        long long ans = 0;
        long long left = 0;
        int n = prices.size();
        while (left < n) {
            long long right = left + 1;
            while (right < n && prices[right] - prices[right - 1] == -1) {
                right++;
            }
            long long m = right - left;
            ans += m * (m + 1) / 2;
            left = right;
        }
        return ans;

    }
};

File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.py
class Solution:
    def getDescentPeriods(self, prices: List[int]) -> int:
        prevPrice = -1
        curStreak = 1
        ans = 0
        for price in prices:
            if price > 0 and prevPrice - price == 1:
                curStreak += 1
                ans += curStreak
            else:
                curStreak = 1
                ans += 1
            prevPrice = price
        return ans